com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/SetBreakpointsRequestHandler.java

```diff
@@ -73,6 +84,11 @@ public void initialize(IDebugAdapterContext context) {
             return AdapterUtils.createAsyncErrorResponse(response, ErrorCode.EMPTY_DEBUG_SESSION, "Empty debug session.");
         }
 
+        if (!registered) {
```

Comment:
Should synchronize the register behavior.

Reply:
Should register behavior happen on the initialization?

Reply:
on initialization, the debug session is not ready, always null in initialize method

Reply:
There is a initialize member of IDebugRequestHandler to handle the initialization. This logic should be there.